server/bin/pbench-restore-and-unpack-tarballs.py

```diff
@@ -0,0 +1,243 @@
+        ],
+        reverse=True,
+    )
+    tbs_cnt = len(tbs_sorted)
+    tbs_left = tbs_cnt
+    tbs_existing = 0
+    tbs_restored = 0
+    for tb_dt, ctrl, tb in tbs_sorted:
+        try:
+            a_tb = archive_p / ctrl / tb.name
+            tbs_left -= 1
+
+            if a_tb.exists():
+                tbs_existing += 1
+                print(
+                    f"Exists {tbs_existing:d} ({tbs_restored:d} restored) of (tbs_cnt:d): {a_tb}",
+                    flush=True,
+                )
+                continue
+
+            # Create controller directory if it doesn't exist
+            ctrl_p = archive_p / ctrl
+            if not options.dry_run:
+                try:
+                    ctrl_p.mkdir()
+                except FileExistsError:
+                    pass
+                else:
+                    print(f"\nCreated controller directory, '{ctrl_p}'", flush=True)
+
+            # Copy tar ball to archive tree, along with its .md5 ...
+            cp_cmd = f"cp -a {tb} {tb}.md5 {ctrl_p}/"
+            print(f"\n{cp_cmd}", flush=True)
+            if not options.dry_run:
+                cp = subprocess.run(cp_cmd, shell=True, stderr=subprocess.STDOUT)
+                if cp.returncode != 0:
+                    print(
+                        f"FAILURE: cp command: '{cp_cmd}': {cp.returncode}, {cp.stdout!r}",
+                        file=sys.stderr,
+                    )
+                    sys.exit(1)
+            # ... and run md5sum check against it
+            md5sum_cmd = f"md5sum --check {a_tb}.md5"
+            print(md5sum_cmd, flush=True)
+            if not options.dry_run:
+                cp = subprocess.run(md5sum_cmd, shell=True, cwd=str(ctrl_p))
+                if cp.returncode != 0:
+                    print(
+                        f"FAILURE: md5sum command: '{md5sum_cmd}': {cp.returncode}, {cp.stdout!r}",
+                        file=sys.stderr,
+                    )
+                    sys.exit(1)
+
+            # Create the symlink recording the tar ball is already backed up
+            backed_up_dir = ctrl_p / "BACKED-UP"
+            if not options.dry_run:
+                backed_up_dir.mkdir(exist_ok=True)
+            backed_up = backed_up_dir / tb.name
+            print(f"ln -s {a_tb} {backed_up}", flush=True)
+            if not options.dry_run:
+                backed_up.symlink_to(a_tb)
+
+            # Create the symlink requesting the tar ball be unpacked
+            to_unpack_dir = ctrl_p / "TO-RE-UNPACK"
+            if not options.dry_run:
+                to_unpack_dir.mkdir(exist_ok=True)
+            to_unpack = to_unpack_dir / tb.name
+            print(f"ln -s {a_tb} {to_unpack}", flush=True)
+            if not options.dry_run:
+                to_unpack.symlink_to(a_tb)
+
+            tbs_restored += 1
+
+            print(
+                f"\nRestored {tbs_restored:d} ({tbs_existing:d} existing) of ({tbs_cnt:d}): {a_tb}\n",
```

Comment:
I find this pair of prints somewhat confusing (the other one being line 150).  I suggest a something like
```
                "\nFile {:d} of {:d}:  {} restored\n".format(
                tbs_restored + tbs_existing, tbs_cnt, a_tb);
```

Reply:
Because I wanted to see how many were acted upon separate from what was NOT acted upon.

Reply:
In the updated code, the message "Exists 1 (3 restored) of (4):" doesn't read well.

It seems like you have two (or three) concepts that you're trying to present:
1. The name of the file just dealt with
2. The progress against the list
3. The running totals of dispositions

The grand totals of dispositions is presented at the end, so perhaps the running totals don't need to be presented along the way?  If you do want to see them run, that's fine, but I suggest adding a few more characters to make things clear:
```
                "\nFile {:d} of {:d} ({:d} restored, {:d} existing, so far), restored:  {}\n".format(
                tbs_restored + tbs_existing, tbs_restored, tbs_existing, tbs_cnt, a_tb);
```
This gives you the file count so far, the total count (so you know where you are), the count of restored and existing files (so you know what's been done so far), and the status and name of the current file.  (And, the poor human reader doesn't have to figure out which two quantities are the running totals and add them together himself.)